### PR TITLE
Updated BatchNorm Decomposition not to fail with dyamic inputs shapes

### PR DESCRIPTION
**Pull request by GlebKazantaev**

### CODE REVIEW

File: inference-engine/src/transformations/src/transformations/batch_norm_decomposition.cpp [hunk 1]
```diff
@@ -24,39 +24,37 @@ ngraph::pass::BatchNormDecomposition::BatchNormDecomposition() {
...
 
-        auto shift_mul = make_shared<opset1::Multiply>(m_mean, opset1::Constant::create(m_input->get_element_type(), Shape{}, {-1}));
+        auto shift_mul = make_shared<opset1::Multiply>(m_mean, opset1::Constant::create(m_input.get_element_type(), Shape{}, {-1}));
         auto shift = make_shared<opset1::Multiply>(scale, shift_mul);
 
         // Expand Scale, Shift, Gamma and Beta to be aligned with layout
-        size_t dims_to_add = m_input->get_shape().size() - 2;
-        Shape gamma_shape = m_gamma->get_shape();
+        size_t dims_to_add = m_input.get_shape().size() - 2;
+        Shape gamma_shape = m_gamma.get_shape();
```
Comment (ilyachur) [inline inference-engine/src/transformations/src/transformations/batch_norm_decomposition.cpp:53]: In general LGTM. Should we also check shapes for `m_gamma` and `m_beta`?

Comment (GlebKazantaev) [reply inference-engine/src/transformations/src/transformations/batch_norm_decomposition.cpp:53]: I actually do not plan merge this PR as its not solving initial problem with dynamic shapes.
